model/container_registry.py

```diff
@@ -81,6 +82,9 @@ def registry_type(self):
     def api_base_url(self):
         return self.raw.get('api_base_url')
 
+    def rotation_cfg_name(self):
+        return self.raw.get('rotation_cfg_name')
```

Comment:
check for registry-type. if it is not GCR/AR, setting this attribute should at least emit a warning (or rather fail). Also, considering the cfg-name is of a different type (-> GCP I assume?), the type should also be configured, rather than being implied (implying type seems okay-ish if "rotation-cfg-element" is of same type as to-be-rotated element).

otoh: how about defining (-> model/base.py) a new type for rotation-cfg. it could be as simple as:
```
@dataclasses.dataclass
class CfgElementReference:
  type_name: str
  element_name: str
  purpose: str | None = None # might be useful for documenting in-place what the cfg-element-reference is used for
```

Reply:
This is a bit fuzzy as we have the generic cfg-type "container_registry" with its subclassification via "registry_type", but also the dedicated cfg-type "gcp".
However, both cfg-types use the very same rotation coding.

Therefore, if a cfg-element of type "container_registry" specifies a "rotation_cfg_name", it is implied that the referenced cfg is also of type "container_registry"
-> no need for explicit cfg-type specification

Reply:
Thus, the only remaining value in introducing a `CfgElementReference` class would be the purpose attribute, however one could also simply inline that as Yaml comment.

Reply:
> check for registry-type. if it is not GCR/AR, setting this attribute should at least emit a warning (or rather fail). 
good point, will add to validation